core/trino-main/src/main/java/io/trino/server/security/oauth2/ScribeJavaOAuth2Client.java

```diff
@@ -39,15 +39,13 @@
         implements OAuth2Client
 {
     private final DynamicCallbackOAuth2Service service;
-    private final Optional<String> audience;
```

Comment:
Providing an audience while requesting an access token is sometimes required. For example, Auth0 requires it when requesting an access token in JWT format. (https://auth0.com/docs/tokens/access-tokens/get-access-tokens#control-access-token-audience)

Reply:
The audience parameter to Auth0 is a non-standard and proprietary operation, additionally and perhaps more importantly performing that operation against Auth0 just to get a JWT is completely improper, especially as with this PR we support opaque tokens. The reason it's improper is that the mechanism expands the scope of the token to services and APIs not managed by Trino, thus issuing a token which has a much broader scope than necessary. The only possible reason for supporting that audience field for Auth0 would be to capture information that the additional audience provided for authorization purposes, but considering that we don't support any such operations currently it's unnecessary.